social-api/src/main/java/social/service/impl/DiscordSocialDataServiceImpl.java

```diff
@@ -1,16 +1,56 @@
 package social.service.impl;
 
+import com.jayway.jsonpath.JsonPath;
+import lombok.extern.log4j.Log4j2;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import social.model.AuthResponse;
+import social.model.SocialType;
 import social.service.SocialDataService;
 
 @Service
 @Qualifier("discord")
+@Log4j2
 public class DiscordSocialDataServiceImpl implements SocialDataService {
+
+    private final SocialType socialType = SocialType.DISCORD;
+    private final String username;
+
+    @Autowired
+    public DiscordSocialDataServiceImpl(@Value("${social.discord.username:geekbeacon}") String username) {
+        this.username = username;
+    }
+
     @Override
     public int getCount() {
-        return 4368;
+        try {
+            HttpClient client = HttpClients.createMinimal();
+            HttpUriRequest request = RequestBuilder.get()
+                .setUri("https://discordapp.com/api/v6/invites/" + username + "?with_counts=true")
```

Comment:
I think this is a bit cleaner when adding parameters, but not a big deal:

  .addParameter("with_counts", "true")

Reply:
It's not a parameter we'll ever change, so I thought it better to not waste processing power assembling a URL that'll always be static anyway.